oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakAuthorizer.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2017-2023, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.server.authorizer;
+
+import io.strimzi.kafka.oauth.common.ConfigException;
+import org.apache.kafka.common.Endpoint;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.acl.AclBinding;
+import org.apache.kafka.common.acl.AclBindingFilter;
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.resource.ResourceType;
+import org.apache.kafka.metadata.authorizer.AclMutator;
+import org.apache.kafka.metadata.authorizer.ClusterMetadataAuthorizer;
+import org.apache.kafka.metadata.authorizer.StandardAcl;
+import org.apache.kafka.metadata.authorizer.StandardAuthorizer;
+import org.apache.kafka.server.authorizer.AclCreateResult;
+import org.apache.kafka.server.authorizer.AclDeleteResult;
+import org.apache.kafka.server.authorizer.Action;
+import org.apache.kafka.server.authorizer.AuthorizableRequestContext;
+import org.apache.kafka.server.authorizer.AuthorizationResult;
+import org.apache.kafka.server.authorizer.AuthorizerServerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+/**
+ * An authorizer that ensures a single stateful instance is used for authorization callbacks.
+ */
+public class KeycloakAuthorizer implements ClusterMetadataAuthorizer {
+
+    private static final Logger log = LoggerFactory.getLogger(KeycloakAuthorizer.class);
+
+    private final static AtomicInteger VERSION_COUNTER = new AtomicInteger(1);
+    private final int version = VERSION_COUNTER.getAndIncrement();
```

Comment:
Wouldn't this be clearer:

```suggestion
    private final static AtomicInteger VERSION_COUNTER = new AtomicInteger(2);
    private final int version = 1;
```

Reply:
The intention of this code is for created instances to be versioned. When in KRaft mode multiple instances of the authorizer can be instantiated. 

Your suggestion would assign them all version == 1, and VERSION_COUNTER is then redundant and it's never used. 

My intention is for each new instance to get the next incremental number. You can then easily track the instances via toString() and also you have some idea how many there have been instantiated. This `version` is not serialization version.

Reply:
I renamed the variables to minimise confusion.